core/src/main/java/org/apache/accumulo/core/logging/TabletLogger.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.accumulo.core.logging;
+
+import static java.util.stream.Collectors.toList;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.accumulo.core.dataImpl.KeyExtent;
+import org.apache.accumulo.core.metadata.schema.Ample;
+import org.apache.accumulo.core.tabletserver.log.LogEntry;
+import org.apache.accumulo.core.util.HostAndPort;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+// logs messages about a tablets internal state, like its location, set of files, metadata... operations that cut across multiple tablets should go elsewhere
+public class TabletLogger {
+
+  private static final String PREFIX = Logging.PREFIX + "tablet.";
+
+  private static final Logger locLog = LoggerFactory.getLogger(PREFIX + "location");
+  private static final Logger fileLog = LoggerFactory.getLogger(PREFIX + "files");
+  private static final Logger recoveryLog = LoggerFactory.getLogger(PREFIX + "recovery");
+  private static final Logger walsLog = LoggerFactory.getLogger(PREFIX + "wals");
+
+  /**
+   * A decision was made to assign a tablet to a tablet server process. Accumulo will stick to this
+   * decision until the tablet server loads the tablet or dies.
+   */
+  public static void assigned(KeyExtent extent, Ample.TServer server) {
+    locLog.debug("Assigned {} to {}", extent, server);
```

Comment:
For all those messages with a KeyExtent, it might be good to put the KeyExtent in the same position in each message, in a more structured way, so it's easier to parse out by log analyzers.

For example, any of these:

```java
locLog.debug("Assigned tablet to server; server={}, tablet={}", server, extent); // description with named params
locLog.debug("action=assignment; server={}, tablet={}", server, extent); // all named params
locLog.debug("{} assigned to server {}", extent, server); // KeyExtent always first strategy
locLog.debug("Tablet({}) assigned to server {}", extent, server); // KeyExtent always first strategy, with descriptive structure
```

Reply:
I would like to experiment with creating grok patterns to parse these messages and use that refine the messages.  Not sure I will get around to that.

Reply:
If the messages are relatively easily parseable with something like LogStash grok patterns, I think it's fine to keep the messages more human-readable. As long as thought is put in to avoid making them too difficult to parse with automation, I'm fine with whatever messages you use.